app/pages/lab/project.jsx

```diff
@@ -0,0 +1,359 @@
+import { cloneElement, useEffect, useState } from 'react'
+import PropTypes from 'prop-types';
+
+import { Link, IndexLink } from 'react-router';
+import LoadingIndicator from '../../components/loading-indicator.jsx';
+import { Helmet } from 'react-helmet';
+import apiClient from 'panoptes-client/lib/api-client';
+import counterpart from 'counterpart';
+import workflowActions from './actions/workflow.js';
+import isAdmin from '../../lib/is-admin.coffee';
+import LabStatus from '../../partials/lab-status.jsx';
+import { isThisProjectUsingFEMLab, FEM_LAB_PREVIEW_HOST } from '../lab-fem/fem-lab-utilities.js';
+
+const DEFAULT_SUBJECT_SET_NAME = 'Untitled subject set'
+const DELETE_CONFIRMATION_PHRASE = 'I AM DELETING THIS PROJECT'
+
+counterpart.registerTranslations('en', {
+  projectLab: {
+    edit: 'Edit'
+  }
+});
+
+function RenderError({
+  error,
+  info
+}) {
+  return (
+    <div>
+      <h1><code>{error.message}</code></h1>
+      <p>{info?.componentStack && <pre>{info.componentStack}</pre>}</p>
+    </div>
+  )
+}
+
+function EditProjectPage({
+  children,
+  location,
+  project
+}) {
+  const [deleteState, setDeleteState] = useState({
+    deletionError: null,
+    deletionInProgress: false
+  });
+
+  function labPath(pathname = '') {
+    return `/lab/${project.id}${pathname}`;
+  }
+
+  function deleteProject() {
+    setDeleteState(oldState => ({ ...oldState, deletionError: null }));
+    const confirmPrompt = `You are about to delete this project and all its data!
+    Enter ${DELETE_CONFIRMATION_PHRASE} to confirm.`;
+    const confirmed = prompt(confirmPrompt) === DELETE_CONFIRMATION_PHRASE;
+
+    if (confirmed) {
+      setDeleteState(oldState => ({ ...oldState, deletionInProgress: true }));
+      project.delete()
+        .then(() => {
+          context.router.push('/lab')
```

Comment:
I think legacy context has been removed in React 17, so this probably won't work any more.